ui/timeline/src/commonMain/kotlin/social/androiddev/timeline/TimelineScreen.kt

```diff
@@ -1,99 +1,46 @@
 package social.androiddev.timeline
 
 import androidx.compose.foundation.background
-import androidx.compose.foundation.layout.Column
-import androidx.compose.foundation.layout.Row
-import androidx.compose.foundation.layout.Spacer
 import androidx.compose.foundation.layout.fillMaxSize
 import androidx.compose.foundation.layout.padding
-import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentSize
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.items
-import androidx.compose.material.Icon
-import androidx.compose.material.MaterialTheme
 import androidx.compose.material.Surface
-import androidx.compose.material.Text
-import androidx.compose.material.icons.Icons
 import androidx.compose.runtime.Composable
 import androidx.compose.ui.Modifier
 import androidx.compose.ui.graphics.Color
 import androidx.compose.ui.unit.dp
-import social.androiddev.common.composables.UserAvatar
 import social.androiddev.common.theme.MastodonTheme
-import social.androiddev.domain.timeline.model.Account
-import social.androiddev.domain.timeline.model.Status
 
 @Composable
 fun TimelineScreen(
+    items: List<FeedItemState>,
     modifier: Modifier = Modifier,
 ) {
-    Surface(modifier = modifier.background(Color(0xFF292C36)).padding(8.dp).fillMaxSize()) {
-        TimelineStatus(status = testStatus1)
-    }
-}
-
-@Composable
-fun TimelineStatus(
-    modifier: Modifier = Modifier,
-    status: Status
-) {
-    // TODO: overflow menu
-    Row {
-        UserAvatar(
-            modifier = Modifier.padding(8.dp),
-            url = status.account.avatar
-        )
-        Column {
-            Row {
-                Text(
-                    text = status.account.username,
-                    color = Color.White
-                )
-                Spacer(modifier = Modifier.width(4.dp))
-                Text(
-                    modifier = Modifier.weight(1f),
-                    text = status.account.acct,
-                    color = Color(0xFF616981)
+    Surface(
+        modifier = modifier
+            .background(Color(0xFF292C36))
+            .padding(8.dp)
+            .fillMaxSize()
+    ) {
+        LazyColumn {
+            items(items, key = { item -> item.id }) { state ->
+                FeedCard(
+                    state = state,
+                    modifier = Modifier.wrapContentSize(),
                 )
-                Row {
-                    Text(
-                        text = status.createdAt,
-                        color = Color(0xFF616981)
-                    )
-                }
             }
-            Text(status.content)
         }
     }
-    // TODO: footer (reply, boost, favorite, share)
```

Comment:
Should we add the actions in here or below it in the parent item?

Reply:
So I figured for MVP we won't support any actions, just Create an account/Log in -> View read-only Local Timeline and the ability to create a Toot (maybe just text initially?)

So I didn't want to add a bunch of TODOs and commented-out code here until we get to it.